frontend/src/views/NftDisplay.vue

```diff
@@ -0,0 +1,339 @@
+import { isNftType, Nft, allNftTypes } from '../interfaces/Nft';
+import { Accessors } from 'vue/types/options';
+import { Contract, Contracts } from '@/interfaces';
+import BigButton from '../components/BigButton.vue';
+import { NftIdType } from '@/components/smart/NftList.vue';
+import { Characters, Shields, Weapons } from '../../../build/abi-interfaces';
+import WeaponGrid from '../components/smart/WeaponGrid.vue';
+import CharacterList from '../components/smart/CharacterList.vue';
+import NftList from '../components/smart/NftList.vue';
+
+interface StoreMappedGetters {
+  contracts: Contracts;
+  weaponsWithIds(ids: (string | number)[]): Nft[];
+  nftsWithIdType(nftIdType: NftIdType[]): Nft[];
+  shieldsWithIds(ids: string[]): Nft[];
+  charactersWithIds(ids: (string | number)[]): Nft[];
+}
+interface StoreMappedActions {
+  fetchShields(shieldIds: (string | number)[]): Promise<void>;
+  fetchWeapons(weaponIds: (string | number)[]): Promise<void>;
+  fetchCharacters(characterIds: (string | number)[]): Promise<void>;
+  checkMarketItemOwnership(payload: {
+    nftContractAddr: string;
+    tokenId: string | number;
+  }): Promise<string>;
+}
+
+export default Vue.extend({
+  props: {
+    nftTypeProp: {
+      type: String,
+      validator(type) {
+        return isNftType(type);
+      },
+    },
+    nftIdProp: {
+      type: String,
+    },
+  },
+
+  data() {
+    return {
+      weapon: null as Nft | null,
+      character: null as Nft | null,
+      shield: null as Nft | null,
+      allNftTypes,
+      ownerAddress: '',
+      nftType: '',
+      nftId: '',
+    };
+  },
+
+  computed: {
+    ...(mapGetters([
+      'contracts',
+      'weaponsWithIds',
+      'nftsWithIdType',
+      'shieldsWithIds',
+      'charactersWithIds',
+    ]) as Accessors<StoreMappedGetters>),
+
+    isKnownNftType(): boolean {
+      return isNftType(this.nftType);
+    },
+
+    Weapons(): Contract<Weapons> {
+      // we use x! here because we assert that they're set already in created()
+      // this helps with typings
+      return this.contracts.Weapons!;
+    },
+
+    Characters(): Contract<Characters> {
+      // we use x! here because we assert that they're set already in created()
+      // this helps with typings
+      return this.contracts.Characters!;
+    },
+
+    Shields(): Contract<Shields> {
+      // we use x! here because we assert that they're set already in created()
+      // this helps with typings
+      return this.contracts.Shields!;
+    },
+
+    contractAddress(): string {
+      return this.nftType === 'weapon'
+        ? this.Weapons.options.address
+        : this.nftType === 'character'
+          ? this.Characters.options.address
+          : this.Shields.options.address;
+    },
+  },
+
+  methods: {
+    ...(mapActions([
+      'fetchShields',
+      'fetchWeapons',
+      'fetchCharacters',
+      'checkMarketItemOwnership',
+    ]) as StoreMappedActions),
+    async displayCharacter(): Promise<void> {
```

Comment:
Why is this async? Similar functions below are not.

Reply:
Woops, left it out, I'll correct it now